### DISCUSSION TITLE
Minimum system requirements for hosting server?

### DISCUSSION THREAD

**Discussion by Sharaddition** [ente-io/ente#594] [Q&A] [answered]
I wish to host ente-photos on very low-end embedded device Khadas Vim1. It has `2GB Ram` & `Amlogic S905X SoC`. Can it handle?

I have tried immich, but since it does on server ml operations, it didn't worked.

**Comment by mnvr** [comment, marked as answer]
There aren't any "minimum" system requirements as such, the server process is very light weight - it's just a single go binary, and it doesn't do any server side ML, so I feel it should be able to run on anything reasonable.

**Reply by Sharaddition** [reply]
Thanks for prompt reply, will try hosting on the device.

**Comment by Sharaddition** [comment]
The installation was butter smooth with docker. It worked perfectly on my Khadas Vim1.
Without any load, it was consuming around `750mb` ram & `1-2%` cpu.

**Reply by mnvr** [reply]
Awesome. And thank you for taking the time to report back your findings!

**Reply by ashutoshsaboo** [reply]
@Sharaddition just curious, were you trying to self host ente auth or ente photos? If latter, what did you use for your storage service (given you mentioned your embedded device has less specs)?

**Reply by Sharaddition** [reply]
@ashutoshsaboo I was hosting ente photos. It was just a test, till photos app supports 3 party servers. Since it was docker installation, I'm planning to mount my spare HDD with docker volumes. I hope it clears your doubt.

**Reply by smutax** [reply]
> @ashutoshsaboo I was hosting ente photos. It was just a test, till photos app supports 3 party servers. Since it was docker installation, I'm planning to mount my spare HDD with docker volumes. I hope it clears your doubt.

Hi. Very interesting. How did you get on with mounting spare hdd with docker volumes? After some reading i got an impression that Ente can do server self-hosting, but user photos are not self-hosted and only mirrored to 3 cloud storage providers. Thanks

**Reply by Sharaddition** [reply]
Hey @ashutoshsaboo, I think there is a little misunderstanding. With just 2 lines of edit I was successfully able to mount my HDD. And all the data is stored in my storage only.

![image_2024-10-30_003719424](https://github.com/user-attachments/assets/2f67f80d-465d-4a39-ac2d-92ce6aa25fe2)
